src/routes/transactions/mappers/multisig-transaction.mapper.ts

```diff
@@ -0,0 +1,592 @@
+    const direction = this.mapTransferDirection(
+      transaction.safe,
+      sender,
+      recipient,
+    );
+
+    const senderAddressInfo =
+      sender === transaction.safe
+        ? { value: sender }
+        : await this.addressInfoHelper.getOrDefault(chainId, sender);
+
+    const recipientAddressInfo =
+      recipient === transaction.safe
+        ? { value: recipient }
+        : await this.addressInfoHelper.getOrDefault(chainId, recipient);
+
+    return <TransferTransaction>{
+      type: 'Transfer',
+      sender: this.filterAddressInfo(senderAddressInfo),
+      recipient: this.filterAddressInfo(recipientAddressInfo),
+      direction,
+      transferInfo: <Erc20TransferInfo>{
+        type: 'ERC20',
+        tokenAddress: token.address,
+        tokenName: token.name,
+        tokenSymbol: token.symbol,
+        logoUri: token.logoUri,
+        decimals: token.decimals,
+        value: this.getValueParam(dataDecoded, '0'),
+      },
+    };
+  }
+
+  private async mapErc721Transfer(
+    token: Token,
+    chainId: string,
+    transaction: MultisigTransaction,
+  ): Promise<TransferTransaction> {
+    const { dataDecoded } = transaction;
+    const sender = this.getFromParam(dataDecoded, transaction.safe);
+    const recipient = this.getToParam(dataDecoded, this.NULL_ADDRESS);
+    const direction = this.mapTransferDirection(
+      transaction.safe,
+      sender,
+      recipient,
+    );
+
+    const senderAddressInfo =
+      sender === transaction.safe
+        ? { value: sender }
+        : await this.addressInfoHelper.getOrDefault(chainId, sender);
+
+    const recipientAddressInfo =
+      recipient === transaction.safe
+        ? { value: recipient }
+        : await this.addressInfoHelper.getOrDefault(chainId, recipient);
+
+    return <TransferTransaction>{
+      type: 'Transfer',
+      sender: this.filterAddressInfo(senderAddressInfo),
+      recipient: this.filterAddressInfo(recipientAddressInfo),
+      direction,
+      transferInfo: <Erc721TransferInfo>{
+        type: 'ERC721',
+        tokenAddress: token.address,
+        tokenId: this.getValueParam(dataDecoded, '0'),
+        tokenName: token.name,
+        tokenSymbol: token.symbol,
+        logoUri: token.logoUri,
+      },
+    };
+  }
+
+  private getActionCount(transaction: MultisigTransaction): number | undefined {
+    const { dataDecoded } = transaction;
+    if (transaction?.dataDecoded?.method === 'multiSend') {
+      const parameter = dataDecoded.parameters?.find(
+        (parameter) => parameter.name === 'transactions',
+      );
+      return parameter?.valueDecoded?.length;
+    }
+  }
+
+  private async getToEtherTransfer(
+    chainId: string,
+    transaction: MultisigTransaction,
+    safe: Safe,
+  ): Promise<TransferTransaction> {
+    const recipient =
+      transaction.safe !== transaction.to
+        ? await this.addressInfoHelper.getOrDefault(chainId, transaction.to)
+        : { value: transaction.to };
```

Comment:
Why do we have this fallback?

Reply:
I just rewrote it from the Rust implementation https://github.com/safe-global/safe-client-gateway/blob/8855fec20cb1ac2cb9783802dad35ba1071578bc/src/common/converters/mod.rs#L25

Maybe I'm missing something and the fallback is not necessary, do you think we could just use `transaction.to` value directly here?

Reply:
I think this is related with some other comments that I made: https://github.com/5afe/safe-client-gateway-nest/pull/197#discussion_r1045949407 and https://github.com/5afe/safe-client-gateway-nest/pull/197#discussion_r1045599136

We should just return here whatever the `addressInfoHelper` returns. Calling `addressInfoHelper.getOrDefault` and then having another default seems a bit strange 🤔

Reply:
I was trying to mimic the exact response from the Rust CGW implementation, even for `nulls`, that's the reason for that defaults, but I agree maybe it doesn't make much sense. I removed all the fallbacks to rely on `AddressInfoHelper`. It's changed in [a70656a](https://github.com/5afe/safe-client-gateway-nest/pull/197/commits/a70656aa38e4762de4339cba81c6ec7ff0d4fd13)